pkg/bootstrap/platform/gcp.go

```diff
@@ -218,7 +272,12 @@ func parseGCPMetadata() (pid, npid, cluster, location string) {
 			}
 		}
 	})
-	return envPid, envNpid, envCluster, envLocation
+
+	if envPid == "" || envNpid == "" || envCluster == "" || envLocation == "" {
```

Comment:
i don't think this is quite right, and will cause issues (at least somewhere). It is entirely possible for someone to use `GCP_METADATA=my-project|||`, etc.

This appears to add a requirement that all evaluate to something that isn't the empty string, if I understand correctly. And the fallback is that if any are empty, then all must be retrieved from the MD server?  I think we still need the support for partial parsing.

Reply:
It makes sense, I reverted the changes to allow for partial parsing. Though this section of the code is very difficult to read.